api/api.go

```diff
@@ -0,0 +1,129 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+)
+
+// WeatherReponse is used to accept JSON structured
+// data.  Output is not very human friendly and is thus
+// a stage type for the Weather struct
+type WeatherResponse struct {
+	Weather []struct {
+		Main        string `json:"main"`
+		Description string `json:"description"`
+	} `json:"weather"`
+	Main struct {
+		Temp float64 `json:"temp"`
+	} `json:"main"`
+	City string `json:"name"`
+}
+
+// Weather is the human friendly struct that is populated
+// by the WeatherReponse struct
+type Weather struct {
+	Main        string
+	Description string
+	Temp        float64
+	City        string
+}
+
+// Client is used to assemble the necessary parts for a HTTP request
+// and includes the HTTPClient lib to set timeout
+type Client struct {
+	Base       string
+	Units      string
+	ApiKey     string
+	HTTPClient *http.Client
+}
+
+// Get takes a location and resturns a Weather struct and error.
+func (c Client) Get(location string) (Weather, error) {
+
+	//assemble request based on Client struct data
+	url := c.FormatURL(location)
+
+	//use Client since we override default HTTPClient settings -- timeout
+	resp, err := c.HTTPClient.Get(url)
+
+	if err != nil {
+		return Weather{}, fmt.Errorf("something went wrong.  Please try again later.  %v", err)
```

Comment:
If you're seeing an error message, it's usually because something went wrong, isn't it? We can often simply report the `http` error directly, but sometimes the value of `fmt.Errorf` can be in giving some context about what we were _doing_ when the error happened. For example:

```suggestion
		return Weather{}, fmt.Errorf("error contacting API: %v", err)
```

Reply:
Yes, makes sense.  Will incorporate revised error message